### PR TITLE
Adding QOS schema changes

### PR DESCRIPTION
**Pull request by hrachya-mellanox**
Contains
- definitions of QOS related ABNF tokens
- samples
- definition for JSON configuration file.

### CODE REVIEW

File: doc/swss-schema.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Schema data is defined in ABNF [RFC5234](https://tools.ietf.org/html/rfc5234) syntax.  
 
+
 ###PORT_TABLE
-Stores information for physical switch ports managed by the switch chip.  device_names are defined in [port_config.ini](../portsyncd/port_config.ini).  Ports to the CPU (ie: management port) and logical ports (loopback) are not declared in the PORT_TABLE.   See INTF_TABLE.
+Stores information for physical switch ports managed by the switch chip.  device_names are defined in [port_config.ini](https://github.com/stcheng/swss/blob/mock/portsyncd/port_config.ini).  Ports to the CPU (ie: management port) and logical ports (loopback) are not declared in the PORT_TABLE.   See INTF_TABLE.
```
Comment (eladraz) [inline doc/swss-schema.md]: Please rebase to latest

File: doc/swss-schema.md [hunk 2]
```diff
@@ -13,6 +14,10 @@ Stores information for physical switch ports managed by the switch chip.  device
     lanes               = list of lanes ; (need format spec???)
     ifname              = 1*64VCHAR     ; name of the port, must be unique 
     mac                 = 12HEXDIG      ; 
+
+
+
+---------
```
Comment (eladraz) [inline doc/swss-schema.md]: Why did you added these changes?

File: doc/swss-schema.md [hunk 3]
```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+				"wred_profile"	: 	"[WRED_PROFILE_TABLE:AZURE]"
+			},
+			"OP": "SET"
+		}
+	]
+
+
+
+
+----------
```
Comment (eladraz) [inline doc/swss-schema.md]: No need for all the extra newlines and '-----'

**Comment by eladraz** [comment]
Please format the commit message as in README.md (Signed-off-by and commit tittle)

File: doc/swss-schema.md [hunk 4]
```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+	queue_index			= number
+	;field values
+	[0-7]
+	; Example
+	; key QUEUE_TABLE:BEST_EFFORT
+	;		field_name				value
+	; "queue_index"					  1
+	; key QUEUE_TABLE:RDMA
+	;		field_name				value
+	; "queue_index"					  4
```
Comment (eladraz) [inline doc/swss-schema.md]: I just want to understand. This table translate queue indexes to names? (e.g. BEST_EFFORT/RDMA/etc?)
If so, why it's needed?

File: doc/swss-schema.md [hunk 5]
```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
 
 
+---------
+##QOS related definitions
+
+###Definitions of common tokens
+	number					= 1*DIGIT
+	name					= number/1*ALPHA
+	ref_hash_key_reference	= "[" hash_key "]" ;Value the token is a refernce to the name of another valid DB key.
+	hash_key				= name ; a valid key name (i.e. exists in DB)
```
Comment (eladraz) [inline doc/swss-schema.md]: I don't think this is needed. Please use the same format as @dplore uses

File: doc/swss-schema.md [hunk 6]
```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+	tc_num				= number
+	;values
+	queue				= ref_hash_key_reference ;refernce to a queue hash key, see example below .
+	
+	; example
+	; key "TC_TO_QUEUE_MAP_TABLE:AZURE"
+	;	field_name					value
+	;		5					[QUEUE_TABLE:BEST_EFFORT]		; resolves to queue 1
+	;		6					[QUEUE_TABLE:BEST_EFFORT]		; resolves to queue 1
+
```
Comment (eladraz) [inline doc/swss-schema.md]: I think that example should be a snapshot of a radis-cli output. It's more clear
(e.g. Why do you need "QUEUE_TABLE:BEST_EFFORT" and just use the queue index?

File: doc/swss-schema.md [hunk 7]
```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+	ref_hash_key_reference	= "[" hash_key "]" ;Value the token is a refernce to the name of another valid DB key.
+	hash_key				= name ; a valid key name (i.e. exists in DB)
+
+###QUEUE_TABLE
+
+	; QUEUE table. Defines types of queues and their IDs to be referenced by the schedulers, ports.
+	; This is a 'template' for a queue[i], which then will be applied to a concrete port queue.
+	; For example can be applied to port.0.queue.3 port.1.queue.3, etc.
+	; SAI mapping - no direct mapping. As mentioned this is a 'template' for a queue.
+	key					= QUEUE_TABLE:name
```
Comment (eladraz) [inline doc/swss-schema.md]: Please use the same formatting as the rest of the document

File: doc/swss-schema.md (hunk 3 above)
Comment (lguohan) [inline doc/swss-schema.md]: I am thinking would it be better to put move scheduler, wred_profile fields from QUEUE_QOS_TABLE into QUEUE_TABLE? It aligns with saiqueue.h definition and also address the other question raised by elad.

File: doc/swss-schema.md (hunk 4 above)
Comment (lguohan) [inline doc/swss-schema.md]: QUEUE_TABLE is similar to the PORT_TABLE. Each entry in the the queue_table should represent a queue instance on a specific port in the system. Then, it also has a set of attributes that is associated with the queue.

File: doc/swss-schema.md (hunk 6 above)
Comment (lguohan) [inline doc/swss-schema.md]: agree that redis-cli format helps

File: doc/swss-schema.md (hunk 3 above)
Comment (dplore) [inline doc/swss-schema.md]: I am ok with the line separators.  But probably only need 1 or 2 whitespaces.

**Comment by lguohan** [comment]
Ack

**Comment by stcheng** [comment]
merged

**Pull request closed without merging**